src/zenml/post_execution/pipeline_run.py

```diff
@@ -147,13 +147,17 @@ def extra(self) -> Dict[str, Any]:
         return cast(Dict[str, Any], extra)
 
     @property
-    def enable_cache(self) -> bool:
+    def enable_cache(self) -> Optional[bool]:
         """Returns whether caching is enabled for this pipeline run.
 
         Returns:
             True if caching is enabled for this pipeline run.
         """
-        enable_cache = self.pipeline_configuration["enable_cache"]
+        from zenml.pipelines.base_pipeline import PARAM_ENABLE_CACHE
+
+        if PARAM_ENABLE_CACHE not in self.pipeline_configuration:
```

Comment:
`return self.pipeline_configuration.get(PARAM_ENABLE_CACHE)` isn't this the same?

Reply:
mhh, yes, no clue why I wrote this convoluted garbage 😅